FlowCrypt/Functionality/Api/Remote Pub Key Apis/PubLookup.swift

```diff
@@ -68,8 +68,13 @@ class PubLookup: PubLookupType {
     }
 
     func fetchRemoteUpdateLocal(with recipient: Recipient) async throws -> RecipientWithSortedPubKeys {
-        let recipient = try await self.lookup(recipient: recipient)
-        try localContactsProvider.updateKeys(for: recipient)
-        return recipient
+        let remoteRecipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
+        if let localContact = try await localContactsProvider.searchRecipient(with: recipient.email), // First check if we have local saved contact
+          localContact.keyState == .revoked, // Check if local contact key is revoked because if local contact key state is revoked, we should not use further keys with same fingerprints. Instead we should use local revoked key from now on
+          localContact.pubKeys == remoteRecipient.pubKeys { // Even local saved key is revoked key, we should use newer(remote) key if fingerprints(pubkeys) are not same
+           return localContact
+        }
+        try localContactsProvider.updateKeys(for: remoteRecipient)
+        return remoteRecipient
     }
```

Comment:
I like the rename - it's clearer. My comment still stands. The `if` branch will only execute when `localContact.pubKeys == remoteRecipient.pubKeys`. If they are equal, why not return `remoteContact`? Therefore it could be rewritten as:

```swift
        let remoteRecipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
        if let localContact = try await localContactsProvider.searchRecipient(with: recipient.email), // First check if we have local saved contact
          localContact.keyState == .revoked, // Check if local contact key is revoked because if local contact key state is revoked, we should not use further keys with same fingerprints. Instead we should use local revoked key from now on
          localContact.pubKeys == remoteRecipient.pubKeys { // Even local saved key is revoked key, we should use newer(remote) key if fingerprints(pubkeys) are not same
           return remoteRecipient // you just checked it's the same
        }
        try localContactsProvider.updateKeys(for: remoteRecipient)
        return remoteRecipient
```

Further, when `localContact.pubKeys == remoteRecipient.pubKeys` then `updateKeys` is a no-op. Nothing to update. Therefore can be rewritten as:

```swift
        let remoteRecipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
        try localContactsProvider.updateKeys(for: remoteRecipient)
        return remoteRecipient
```

Reply:
There's exception when local contact key state is revoked.
Let's say you have revoked key in local and you receive updated key (with same fingerprint)
With below code, app will try to display updated key info. (In this case, we should display revoked key information to user as same fingerprint key should never be replaced)
Let me know if it makes sense.

```
let remoteRecipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
try localContactsProvider.updateKeys(for: remoteRecipient)
return remoteRecipient
```
> The if branch will only execute when localContact.pubKeys == remoteRecipient.pubKeys. If they are equal, why not return remoteContact?

Reply:
>  Let's say you have revoked key in local and you receive updated key (with same fingerprint)

If received public keys are different from local public keys (received is valid and local is revoked), then `localContact.pubKeys == remoteRecipient.pubKeys` **will be false** and this branch will not execute. You will and up executing the following:

```swift
        try localContactsProvider.updateKeys(for: remoteRecipient)
        return remoteRecipient
```